pkg/cmd/codespace/ssh.go

```diff
@@ -181,85 +190,105 @@ func (a *App) printOpenSSHConfig(ctx context.Context, opts configOptions) error
 		return fmt.Errorf("error getting codespace info: %w", err)
 	}
 
-	t, err := template.New("ssh_config").Parse(`Host cs.{{.Name}}.{{.EscapedRef}}
-	User {{.SSHUser}}
-	ProxyCommand {{.GHExec}} cs ssh -c {{.Name}} --stdio
-	UserKnownHostsFile=/dev/null
-	StrictHostKeyChecking no
-	LogLevel quiet
-	ControlMaster auto
-
-`)
-	if err != nil {
-		return fmt.Errorf("error formatting template: %w", err)
-	}
-
-	ghexec, err := os.Executable()
-	if err != nil {
-		return err
-	}
-
-	// store a mapping of repository -> remote ssh username. This is
-	// necessary because the username can vary between codespaces, but
-	// since fetching it is slow, we store it here so we at least only do
-	// it once per repository.
-	sshUsers := map[string]string{}
-
+	sshUsers := make(chan sshResult)
+	fetches := 0
+	var status error
 	for _, cs := range codespaces {
-
-		if cs.State != "Available" {
+		if cs.State != "Available" && opts.codespace == "" {
 			fmt.Fprintf(os.Stderr, "skipping unavailable codespace %s: %s\n", cs.Name, cs.State)
+			status = cmdutil.SilentError
 			continue
 		}
 
-		sshUser, ok := sshUsers[cs.Repository.FullName]
-		if !ok {
-			codespace, err := a.apiClient.GetCodespace(ctx, cs.Name, true)
-			if err != nil {
-				return fmt.Errorf("getting full codespace details: %w", err)
-			}
+		cs := cs
+		fetches += 1
+		go func() {
+			result := sshResult{}
+			defer func() {
+				select {
+				case sshUsers <- result:
+				case <-ctx.Done():
+				}
+			}()
 
-			session, err := openSSHSession(ctx, a, codespace, nil)
+			session, err := a.openSSHSession(ctx, cs, nil)
 			if err != nil {
-				fmt.Fprintf(os.Stderr, "error connecting to codespace: %v\n", err)
-
-				// Move on to the next codespace. We don't want to bail here - just because we're not
-				// able to set up connectivity to one doesn't mean we shouldn't make a best effort to
-				// generate configs for the rest of them.
-				continue
+				result.err = fmt.Errorf("error connecting to codespace: %w", err)
+				return
 			}
 			defer session.Close()
 
-			a.StartProgressIndicatorWithLabel(fmt.Sprintf("Fetching SSH Details for %s", cs.Name))
-			_, sshUser, err = session.StartSSHServer(ctx)
-			a.StopProgressIndicator()
+			//a.StartProgressIndicatorWithLabel(fmt.Sprintf("Fetching SSH Details for %s", cs.Name))
+			_, result.user, err = session.StartSSHServer(ctx)
+			//a.StopProgressIndicator()
 			if err != nil {
-				fmt.Fprintf(os.Stderr, "error getting ssh server details: %v", err)
-				continue // see above
+				result.err = fmt.Errorf("error getting ssh server details: %w", err)
+				return
 			}
-			sshUsers[cs.Repository.FullName] = sshUser
+
+			result.codespace = cs
+		}()
+	}
+
+	// While the above fetches are running, ensure that the user has keys installed.
+	// That lets us report a more useful error message if they don't.
+	if err = a.ensureAuthorizedKeys(ctx); err != nil {
+		return err
+	}
+
+	ghexec, err := os.Executable()
```

Comment:
Let's use the result of `factory.Executable()` instead of `os.Executable()` here. The former accounts for package managers and symlinks that move around with upgrades. The "factory" object is passed as argument when constructing a `cobra.Command`